crates/wasi-common/build.rs

```diff
@@ -20,19 +20,10 @@ mod wasm_tests {
     use std::process::{Command, Stdio};
 
     pub(crate) fn build_and_generate_tests() {
-        // Validate if any of test sources are present and if they changed
-        // This should always work since there is no submodule to init anymore
-        let bin_tests = std::fs::read_dir("wasi-misc-tests/src/bin").unwrap();
-        for test in bin_tests {
-            if let Ok(test_file) = test {
-                let test_file_path = test_file
-                    .path()
-                    .into_os_string()
-                    .into_string()
-                    .expect("test file path");
-                println!("cargo:rerun-if-changed={}", test_file_path);
-            }
-        }
+        // Re-run if there any changes in wasi-misc-tests
+        println!("cargo:rerun-if-changed=wasi-misc-tests");
+        // Ensure we track *all* subdirs and files
+        println!("cargo:rerun-if-changed=wasi-misc-tests/*");
```

Comment:
FWIW this doesn't work, and depending on the directory itself typically won't have the intended behavior. For now the "best" way (which isn't great) is to have what it had before.

Reply:
Hmm, I'll revert to it then. Any ideas how to check if we've added a new file? Would `println!("cargo:rerun-if-changed=wasi-misc-tests/src/bin")` work?